src/main/java/com/iota/iri/storage/LocalSnapshotsPersistenceProvider.java

```diff
@@ -45,7 +45,10 @@ public Boolean saveBatch(List<Pair<Indexable, Persistable>> models) throws Excep
      * @see PersistenceProvider#save(Persistable, Indexable)
      */
     public Boolean save(Persistable model, Indexable index) throws Exception {
-        return provider.save(model, index);
+        if(provider.getClass().desiredAssertionStatus()) {
```

Comment:
It was helping to prevent the segfault I was talking about the other day by ensuring that the instance of the provider was available before executing the save. But it's no longer needed now that i've changed the provider and injected through the `MainInjectionConfiguration` as the provider and `SpentAddressesProviderImpl` are shut down simultaneously.